doc/srv6/srv6_vpn.md

```diff
@@ -0,0 +1,284 @@
+# SRv6 VPN HLD
+
+# Overview
+
+Alibaba is deploying SRv6 VPN with SONiC based white box router at its edge of the network. The Alibaba's SRv6 VPN use case has been described in muliple conferences. The basic idea follows RFC 8986, which we use SRv6 as a network programming framework for applications.  VPN information is used as service anchor points for identifying service instances in the service plane. SRv6 policy could be used to provide paths to let applications reach service instances. The following diagram provides a summary on what Alibaba would like to achieve via SRv6 VPN.
+
+![image](images/1PGN.png)
+
+To help promoting SRv6's adopting in the industry, Alibaba team decides to upstream its deployed SRv6 VPN solution to SONiC community via 202305 release. We expect more contributors would join us to complete SRv6 VPN related features.
+
+# Design
+
+The SONiC SRv6 VPN deployable solution  requires changes from the following three communities.
+
+## FRR
+
+Besides recent FRR changes for SRv6 VPN contributed by Cisco team, Alibaba team needs to bring in the following FRR changes for a deployable solution. 
+
+1.  VRF routes leaking and performance tuning
```

Comment:
Is this really SRv6 specific? or are you capturing vrf leaking for routing in general?

Reply:
It is mentioned above, it is a required feature to be a deployable solution, not SRv6 specific. 

I would prefer community would also focus on building deployable solutions, instead of piling up technologies. These changes would go via FRR's commit.

Reply:
I think, we should treat VRF route leaking as the separate feature and create the new HLD and link that HLD with SRv6 as one of the use-case.

Reply:
@venkatmahalingam  yes. VRF route leaking would have its own HLD in FRR code base, not in SONiC. I just mention it here as it is a required feature to be a deployable solution.

Reply:
Ok, thanks!